File: KI304.Slobodian.Lab2/src/Printer.java
import java.io.File;
import java.io.FileNotFoundException;
import java.io.PrintWriter;
/**
* Class <code>Printer</code> implements the printer
*/
public class Printer {
    private Ink ink;
    private InternetAdapter internetAdapter;
    private PaperTray paperTray;

    private PrintWriter file;

    public boolean isEnabled;
    /**
    * Constructor
    */
    public Printer() throws FileNotFoundException{
        this.ink = new Ink();
        this.internetAdapter = new InternetAdapter();
        this.paperTray = new PaperTray();

        this.isEnabled = false;

        this.file = new PrintWriter(new File("Log.txt"));
    }
    /**
    * Constructor
    *@param <code>redP</code> The persentage of red ink
	*@param <code>greenP</code> The persentage of red ink
    *@param <code>blueP</code> The persentage of red ink
    *@param <code>CType</code> index of the connection type
    *@param <code>IsC</code> bool, if connected is true
    *@param <code>CStrength</code> strength of the connection
    *@param <code>ssid</code> ssid, if WiFi
    *@param <code>pass</code> password, if WiFi
    *@param <code>ptIndex</code> index of the paper type
    *@param <code>pAmount</code> amount of paper in the tray
    *@param <code>maxPAmount</code> maximal amount of paper in the tray
    */
    public Printer(float redP, float greenP, float blueP, int CType, boolean IsC, int CStrength, String ssid, String pass, int ptIndex, int pAmount, int maxPAmount) throws FileNotFoundException{
        this.ink = new Ink(redP, greenP, blueP);
        this.internetAdapter = new InternetAdapter(CType, IsC, CStrength, ssid, pass);
        this.paperTray = new PaperTray(ptIndex, pAmount, maxPAmount);

        this.isEnabled = false;

        this.file = new PrintWriter(new File("Log.txt"));
    }
    /**
	 * Method turns the printer on
	 */
    public void turnOn() {
        isEnabled = true;

        file.println("You succesfully turned your printeron. Great job, champ!");
        file.println();
        file.flush();
    }
    /**
	* Method turns the printer off
	*/
    public void turnOff() {
        isEnabled = false;    

        file.println(internetAdapter.disconnect());
        file.println("You succesfully turned your printer off.");
        file.println();
        file.flush();
    }
    /**
	* Method gets information about inks of the printer
	*/
    public void getInkInfo(){
        file.println("Ink levels of your printer are:");
        file.println("Red: " + ink.checkRedInkLevel()+"%");
        file.println("Green: "+ ink.checkGreenInkLevel()+"%");
        file.println("Blue: "+ ink.checkBlueInkLevel()+"%");
        file.println();
        file.flush();
    }
    /**
	* Method gets information about internet connection of the printer
	*/
    public void getInternetInfo(){
        if(internetAdapter.checkIfConnected()){
            file.println("You are connected to the internet via " + internetAdapter.checkConnectionType()+".");
            if(internetAdapter.checkWifiSSID() != ""){
                file.println("SSID: "+internetAdapter.checkWifiSSID());
                file.println("Password: "+internetAdapter.checkWifiPassword());
            }
            file.println("Connection strength: "+internetAdapter.checkConnectionStrength());
        }
        else{
            file.println("You are not connected to internet.");
        }
        file.println();
        file.flush();
    }
    /**
	* Method gets information about the paper tray of the printer
	*/
    public void getPaperTrayInfo(){
        file.println("As of now, in your paper tray rests " + paperTray.checkPaperAmount() + " sheets of paper out of " + paperTray.checkMaxPaperAmount() + " possible.");
        file.println("The format of the paper is " + paperTray.checkPaperType() + ".");
        file.println();
        file.flush();
    }
    /**
	* Method fills up printer to the brim
	*/
    public void hastyFillUp(){
        file.println("You decided to try and place into your printer maximal amounts of ink and paper, no matter the consequences.");
        file.println("...");
        file.println(ink.fillRedInk(100));
        file.println("...");
        file.println(ink.fillGreenInk(100));
        file.println("...");
        file.println(ink.fillBlueInk(100));
        file.println("...");
        file.println("...");
        file.println(paperTray.changePaperAmount(paperTray.checkMaxPaperAmount()));
        file.println();
        file.flush();
    }
    /**
	* Method fills up printer with set amount of paper and inks
    *@param rAmount amount of red ink
    *@param gAmount amount of green ink
    *@param bAmount amount of blue ink
    *@param pTypeIndex index of the paper type
    *@param pAmount amount of the paper
	*/
    public void toughtfulFillUp(float rAmount, float gAmount, float bAmount, int pTypeIndex, int pAmount){
        file.println("You decided to try and place into your printer ink and paper, paying attention to what are you doing.");
        file.println("...");
        file.println(ink.fillRedInk(rAmount));
        file.println("...");
        file.println(ink.fillGreenInk(gAmount));
        file.println("...");
        file.println(ink.fillBlueInk(bAmount));
        file.println("...");
        file.println(paperTray.changePaperType(pTypeIndex));
        file.println("...");
        file.println(paperTray.changePaperAmount(pAmount));
        file.println();
        file.flush();
    }
    /**
	* Method connects printer tto the internet, either via WiFi or Ethernet
    *@param ssid SSID, if WiFi
    *@param pass Password, if WiFi
    *@param connectionTypeInex index of the connection type
	*/
    public void connectPrinterToDaInternet(String ssid, String pass, int connectionTypeInex){
        file.println("You decided to try and connect your printer to the internet.");
        file.println("...");
        if(connectionTypeInex == 1){
            file.println(internetAdapter.connectToWifi(ssid, pass));
        }
        else if(connectionTypeInex == 2){
            file.println(internetAdapter.connectToEthernet());
        }
        file.println();
        file.flush();
    }
    /**
	* Method tries to print something via internet
    * @param sheets amount of printed sheets
	*/
    public void printViaInternet(int sheets){
        file.println("You decided to try to print some papers on your printer via the internet.");
        file.println("...");
        if(internetAdapter.checkIfConnected()){
            if(paperTray.checkPaperAmount() < sheets || ink.checkRedInkLevel() < sheets || ink.checkGreenInkLevel() < sheets || ink.checkBlueInkLevel() < sheets){
                file.println("You failed! Your printer dont have something!");
            }
            else{
                file.println("Your prints came out nice.");
                file.println(paperTray.changePaperAmount(-sheets));
            }
        }
        else{
           file.println("You failed! Your printer is not connected to da internet!"); 
        }
        file.println();
        file.flush();
    }
    /**
	* Method tries to print something from the Flash drive
    * @param sheets amount of printed sheets
	*/
    public void printViaFlsh(int sheets){
        file.println("You decided to try to print some papers on your printer via Flash drive.");
        file.println("...");
        if(paperTray.checkPaperAmount() < sheets || ink.checkRedInkLevel() < sheets || ink.checkGreenInkLevel() < sheets || ink.checkBlueInkLevel() < sheets){
            file.println("You failed! Your printer dont have something!");
        }
        else{
            file.println("Your prints came out nice.");
            file.println(paperTray.changePaperAmount(-sheets));
        }
        file.println();
        file.flush();
    }
    public void dispose() {
        file.close();
    }
}

File: src/printer.py
from src.ink import Ink
from src.internetadapter import InternetAdapter
from src.papertray import PaperTray

class Printer:
    def __init__(self):
        self.ink = Ink()
        self.internetAdapter = InternetAdapter()
        self.paperTray = PaperTray()
        self.isEnabled = False

        self.file = open("Log.txt", "w")

    def __init__(self, redP, greenP, blueP, CType, IsC, CStrength, ssid, passw, ptIndex, pAmount, maxPAmount):
        self.ink = Ink(redP, greenP, blueP)
        self.internetAdapter = InternetAdapter(CType, IsC, CStrength, ssid, passw)
        self.paperTray = PaperTray(ptIndex, pAmount, maxPAmount)
        self.isEnabled = False

        self.file = open("Log.txt", "w")

    def turnOn(self):
        self.isEnabled = True
        self.file.write("You successfully turned your printer on. Great job, champ!\n\n")
        self.file.flush()

    def turnOff(self):
        self.isEnabled = False
        self.file.write(self.internetAdapter.disconnect() + "\n")
        self.file.write("You successfully turned your printer off.\n\n")
        self.file.flush()

    def getInkInfo(self):
        self.file.write("Ink levels of your printer are:\n")
        self.file.write(f"Red: {self.ink.checkRedInkLevel()}%\n")
        self.file.write(f"Green: {self.ink.checkGreenInkLevel()}%\n")
        self.file.write(f"Blue: {self.ink.checkBlueInkLevel()}%\n\n")
        self.file.flush()

    def getInternetInfo(self):
        if self.internetAdapter.checkIfConnected():
            self.file.write(f"You are connected to the internet via {self.internetAdapter.checkConnectionType()}.\n")
            if self.internetAdapter.checkWifiSSID() != "":
                self.file.write(f"SSID: {self.internetAdapter.checkWifiSSID()}\n")
                self.file.write(f"Password: {self.internetAdapter.checkWifiPassword()}\n")
            self.file.write(f"Connection strength: {self.internetAdapter.checkConnectionStrength()}\n")
        else:
            self.file.write("You are not connected to the internet.\n\n")
        self.file.flush()

    def getPaperTrayInfo(self):
        self.file.write(f"As of now, in your paper tray rests {self.paperTray.checkPaperAmount()} sheets of paper out of {self.paperTray.checkMaxPaperAmount()} possible.\n")
        self.file.write(f"The format of the paper is {self.paperTray.checkPaperType()}.\n\n")
        self.file.flush()

    def hastyFillUp(self):
        self.file.write("You decided to try and place into your printer maximal amounts of ink and paper, no matter the consequences.\n...\n")
        self.file.write(self.ink.fillRedInk(100) + "\n...\n")
        self.file.write(self.ink.fillGreenInk(100) + "\n...\n")
        self.file.write(self.ink.fillBlueInk(100) + "\n...\n...\n")
        self.file.write(self.paperTray.changePaperAmount(self.paperTray.checkMaxPaperAmount()) + "\n\n")
        self.file.flush()

    def thoughtfulFillUp(self, rAmount, gAmount, bAmount, pTypeIndex, pAmount):
        self.file.write("You decided to try and place into your printer ink and paper, paying attention to what you are doing.\n...\n")
        self.file.write(self.ink.fillRedInk(rAmount) + "\n...\n")
        self.file.write(self.ink.fillGreenInk(gAmount) + "\n...\n")
        self.file.write(self.ink.fillBlueInk(bAmount) + "\n...\n...\n")
        self.file.write(self.paperTray.changePaperType(pTypeIndex) + "\n...\n")
        self.file.write(self.paperTray.changePaperAmount(pAmount) + "\n\n")
        self.file.flush()

    def connectPrinterToDaInternet(self, ssid, passw, connectionTypeIndex):
        self.file.write("You decided to try and connect your printer to the internet.\n...\n")
        if connectionTypeIndex == 1:
            self.file.write(self.internetAdapter.connectToWifi(ssid, passw) + "\n\n")
        elif connectionTypeIndex == 2:
            self.file.write(self.internetAdapter.connectToEthernet() + "\n\n")
        self.file.flush()

    def printViaInternet(self, sheets):
        self.file.write("You decided to try to print some papers on your printer via the internet.\n...\n")
        if self.internetAdapter.checkIfConnected():
            if (
                self.paperTray.checkPaperAmount() < sheets
                or self.ink.checkRedInkLevel() < sheets
                or self.ink.checkGreenInkLevel() < sheets
                or self.ink.checkBlueInkLevel() < sheets
            ):
                self.file.write("You failed! Your printer doesn't have something!\n")
            else:
                self.file.write("Your prints came out nice.\n")
                self.file.write(self.paperTray.changePaperAmount(-sheets) + "\n")
        else:
            self.file.write("You failed! Your printer is not connected to da internet!\n")
        self.file.write("\n")
        self.file.flush()

    def printViaFlsh(self, sheets):
        self.file.write("You decided to try to print some papers on your printer via Flash drive.\n...\n")
        if (
            self.paperTray.checkPaperAmount() < sheets
            or self.ink.checkRedInkLevel() < sheets
            or self.ink.checkGreenInkLevel() < sheets
            or self.ink.checkBlueInkLevel() < sheets
        ):
            self.file.write("You failed! Your printer doesn't have something!\n")
        else:
            self.file.write("Your prints came out nice.\n")
            self.file.write(self.paperTray.changePaperAmount(-sheets) + "\n")
        self.file.write("\n")
        self.file.flush()

    def dispose(self):
        self.file.close()
